gdx/src/com/badlogic/gdx/files/FileHandle.java

```diff
@@ -416,25 +401,48 @@ public void writeBytes (byte[] bytes, int offset, int length, boolean append) {
-		String[] relativePaths = file().list();
-		if (relativePaths == null) return new FileHandle[0];
-		FileHandle[] handles = new FileHandle[relativePaths.length];
-		for (int i = 0, n = relativePaths.length; i < n; i++)
-			handles[i] = child(relativePaths[i]);
-		return handles;
-	}
-
-	/** Returns the paths to the children of this directory that satisfy the specified filter. Returns an empty list if this file
-	 * handle represents a file and not a directory. On the desktop, an {@link FileType#Internal} handle to a directory on the
-	 * classpath will return a zero length array.
-	 * @param filter the {@link FileFilter} to filter files
-	 * @throws GdxRuntimeException if this file is an {@link FileType#Classpath} file. */
+	 /**
+	  * Returns the paths to the children of this directory. Returns an empty list if this file handle represents a file and not a
+	  * directory. An {@link FileType#Internal} handle to a directory on the classpath will search for
+	  * an asset.index file in the * root of the asset path. This index should be generated at compile time by a gradle task.
+	  * See for reference: * https://gist.github.com/abueide/e86ea0111025233fb6493dfbcc46501e
+	  *
+	  * @throws GdxRuntimeException if this file is an {@link FileType#Classpath} file.
+	  */
+	 public FileHandle[] list () {
+		  if (type == FileType.Classpath)
+				throw new GdxRuntimeException("Cannot list a classpath directory: " + file);
+		  if (type == FileType.Internal) {
+				if (!Gdx.files.internal("assets.index").exists())
+					 //How is this supposed to be done? Is there a framework level logging system to log a warning?
+					 Gdx.app.log("WARNING: ","Asset index not found, please update your gradle configuration "
+						 + "according to: https://gist.github.com/abueide/e86ea0111025233fb6493dfbcc46501e");
+				else {
+					 String[] filePaths = Gdx.files.internal("assets.index").readString().split("\n");
+					 FileHandle[] files = new FileHandle[filePaths.length];
+					 for (int i = 0; i < filePaths.length; i++) {
+						  files[i] = Gdx.files.internal(filePaths[i]);
+					 }
+					 return files;
+				}
+		  }
+		  String[] relativePaths = file().list();
+		  if (relativePaths == null)
+				return new FileHandle[0];
+		  FileHandle[] handles = new FileHandle[relativePaths.length];
+		  for (int i = 0, n = relativePaths.length; i < n; i++)
+				handles[i] = child(relativePaths[i]);
+		  return handles;
+	 }
+
+	 /**
+	  * Returns the paths to the children of this directory that satisfy the specified filter. Returns an empty list if this file
+	  * handle represents a file and not a directory. An {@link FileType#Internal} handle to a directory on the classpath will
+	  * search for an asset.index file in the root of the asset path. This index should be generated at compile time by a gradle task.
+	  * See for reference: * https://gist.github.com/abueide/e86ea0111025233fb6493dfbcc46501e
+	  *

```

Comment:
These `list` overloads don't need full documentation, I suggest just adding `@see #list() Internal file handling`, or something like that.